wallet-new/src/Cardano/Wallet/Kernel/DB/Sqlite.hs

```diff
@@ -0,0 +1,562 @@
+           Left _  -> returnError Sqlite.ConversionFailed f "not a valid Address"
+           Right a -> pure a
+
+instance FromBackendRow Sqlite Core.Address
+
+
+-- | Creates new 'DatabaseSettings' for the 'MetaDB', locking the backend to
+-- be 'Sqlite'.
+metaDB :: DatabaseSettings Sqlite MetaDB
+metaDB = unCheckDatabase (evaluateDatabase migrateMetaDB)
+
+-- | 'DataType' declaration to convince @Beam@ treating 'Core.Address'(es) as
+-- varchars of arbitrary length.
+address :: DataType SqliteDataTypeSyntax Core.Address
+address = DataType (varCharType Nothing Nothing)
+
+-- | 'DataType' declaration to convince @Beam@ treating 'Core.Timestamp'(s) as
+-- varchars of arbitrary length.
+timestamp :: DataType SqliteDataTypeSyntax Core.Timestamp
+timestamp = DataType (varCharType Nothing Nothing)
+
+-- | 'DataType' declaration to convince @Beam@ treating 'Core.TxId(s) as
+-- varchars of arbitrary length.
+txId :: IsSql92DataTypeSyntax syntax => DataType syntax Core.TxId
+txId = DataType (varCharType Nothing Nothing)
+
+-- | 'DataType' declaration to convince @Beam@ treating 'Core.Coin(s) as
+-- SQL @INTEGER@s.
+coin :: DataType SqliteDataTypeSyntax Core.Coin
+coin = DataType intType
+
+-- | Beam's 'Migration' to create a new 'MetaDB' Sqlite database.
+initialMigration :: () -> Migration SqliteCommandSyntax (CheckedDatabaseSettings Sqlite MetaDB)
+initialMigration () = do
+    MetaDB <$> createTable "tx_metas"
+                 (TxMeta (field "meta_id" txId notNull unique)
+                         (field "meta_amount" coin notNull)
+                         (field "meta_created_at" timestamp notNull)
+                         (field "meta_is_local" boolean notNull)
+                         (field "meta_is_outgoing" boolean notNull))
+           <*> createTable "tx_metas_inputs"
+                 (TxInput (TxCoinDistributionTable (field "input_address" address notNull)
+                                                   (field "input_coin" coin notNull)
+                                                   (TxIdPrimKey (field "meta_id" txId notNull))
+                          ))
+           <*> createTable "tx_metas_outputs"
+                 (TxOutput (TxCoinDistributionTable (field "output_address" address notNull)
+                                                    (field "output_coin" coin notNull)
+                                                    (TxIdPrimKey (field "meta_id" txId notNull))
+                           ))
+
+--- | The full list of migrations available for this 'MetaDB'.
+-- For a more interesting migration, see: https://github.com/tathougies/beam/blob/d3baf0c77b76b008ad34901b47a818ea79439529/beam-postgres/examples/Pagila/Schema.hs#L17-L19
+migrateMetaDB :: MigrationSteps SqliteCommandSyntax () (CheckedDatabaseSettings Sqlite MetaDB)
+migrateMetaDB = migrationStep "Initial migration" initialMigration
+
+
+-- | Migrates the 'MetaDB', potentially mangling the input database.
+-- TODO(adinapoli): Make it safe.
```

Comment:
Maybe don't worry about making it safe here. This SQL migration probably won't live here nor use the beam stuff anyway.